src/app/common/account-restoration/account-restore.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { recurseAccountsForActivity } from './account-restore';
+
+describe(recurseAccountsForActivity.name, () => {
+  test('case of account with activity at 10th index only', async () => {
+    const mockHasActivityFn = vi.fn().mockImplementation((index: number) => {
+      if (index === 10) return true;
+      return false;
+    });
+    const result = await recurseAccountsForActivity({
+      doesAddressHaveActivityFn: mockHasActivityFn,
+    });
+    expect(result).toBe(10);
+    // Index + Min accounts to check + 1
+    expect(mockHasActivityFn.mock.calls.length).toBe(31);
```

Comment:
Trying to understand this, but if true is found at the 10 index why is it the minimum plus 10? Wouldn't the 10th index fall within the minimum number of accounts (20)?

Reply:
If the 10th index (11th account in non index numbers) has a balance, we'd check 0–10, then check another 20 to see if they're empty.

11 + 20

Reply:
Got it! Thanks.